daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -335,7 +336,8 @@ object SExpr {
     */
   final case class SECatch(body: SExpr, handler: SExpr, fin: SExpr) extends SExpr {
     def execute(machine: Machine): Unit = {
-      machine.pushKont(KCatch(handler, fin, machine.frame, machine.actuals, machine.env.size))
+      machine.pushKont(
+        KCatch(machine.markBase(), handler, fin, machine.frame, machine.actuals, machine.env.size))
```

Comment:
This line scares the crap out of me. `machine.markBase()` changes components of the machine related the to environment size and `machine.env.size` reads something related to that. I know that they don't interact _now_ but I'm still scared. If they would interact, the semantics would depend on the order in which arguments are evaluated. I'd be more comfortable if we made the intended evaluation order explicit using `val` bindings.

Also, why do we need to store both `machine.markBase()` and `machine.env.size` in the continuation? That might deserve a comment.

Reply:
I added a comment.

I have a future cleanup PR in prep which relocates all calls to `markBase` to the definition code of the continuations. It makes `val` bindings as you suggest, and is much nicer!

Reply:
Nice. Thank you.